### PR TITLE
feat: support apt proxy

### PR DESCRIPTION
**Pull request by viceice**
Allow passing `APT_HTTP_PROXY` to configure a temporary apt proxy while building buildpack based images.

CI is using a local proxy to speedup parallel and multiple downloads.
Maybe we can use github cache to speedup more in a later pr.

### CODE REVIEW

**Comment by rarkins** [comment]
Will this also work as a build arg for end users who have a permanent apt proxy and want to point the buildpack build to it?

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -34,6 +35,9 @@ jobs:
     steps:
       - uses: actions/checkout@v2.3.4
 
+      - name: start apt-cacher
+        run: docker buildx build --progress plain -t apt-cacher -f Dockerfile.apt-cacher . && docker run -d -p 3142:3142 apt-cacher
```
Comment (rarkins) [inline .github/workflows/build.yaml:39]: Will the cache be reused across all jobs or only within-jobs?

File: Dockerfile.apt-cacher [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM ubuntu:focal@sha256:cf31af331f38d1d7158470e095b132acd126a7180a54f263d386da88eb681d93
+
+VOLUME ["/var/cache/apt-cacher-ng"]
+RUN    apt-get update && apt-get install -y apt-cacher-ng
+
+EXPOSE 3142
+CMD    chmod 777 /var/cache/apt-cacher-ng && /etc/init.d/apt-cacher-ng start && tail -f /var/log/apt-cacher-ng/*
+
+RUN cat /etc/hosts
```
Comment (rarkins) [inline Dockerfile.apt-cacher:9]: Strange that nothing "off the shelf" already existed? Seems like a useful feature

**Review by rarkins — COMMENTED** [review]
Add documentation for end users?

File: Dockerfile.apt-cacher (hunk 2 above)
Comment (viceice) [reply Dockerfile.apt-cacher:9]: yes, no official image, don't want to use unknown image, we can move this to a new repo if you like the general approach.

We can also install and run apt-cacher on ubuntu worker, so no need for an docker image

File: .github/workflows/build.yaml (hunk 1 above)
Comment (viceice) [reply .github/workflows/build.yaml:39]: currently only for the current job. can be extended later to be more persistent

File: Dockerfile.apt-cacher (hunk 2 above)
Comment (viceice) [reply Dockerfile.apt-cacher]: now using native ubuntu package

File: README.md [hunk 3]
```diff
@@ -7,6 +7,11 @@
 
 This repository is the source for the Docker Hub image `containerbase/buildpack`. Commits to `main` branch are automatically built and published.
 
+## Apt proxy
+
+You can pass a custom temporary Apt proxy at build or runtime when installing new packages via `APT_HTTP_PROXY` arg.
```
Comment (rarkins) [inline README.md:12]: Do we support any runtime apt use? Assuming the user doesn't run as root

**Review by rarkins — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (viceice) [reply README.md:12]: Only if the user is root. 😉

**Pull request merged**